test/TestCaseReader.h

```diff
@@ -56,16 +62,35 @@ class TestCaseReader
 	size_t sizetSetting(std::string const& _name, size_t _defaultValue);
 	std::string stringSetting(std::string const& _name, std::string const& _defaultValue);
 
+	template <typename E>
+	E enumSetting(std::string const& _name, std::map<std::string, E> const& _choices, std::string const& _defaultChoice);
+
 	void ensureAllSettingsRead() const;
 
 private:
 	std::pair<SourceMap, std::size_t> parseSourcesAndSettingsWithLineNumber(std::istream& _file);
 	static std::string parseSimpleExpectations(std::istream& _file);
 
+	static void fail(std::string const& _message);
+
```

Comment:
`enumSetting()` being a template pulls a lot of stuff into the header unfortunately. I created this helper mostly to avoid having to use the boost throw macro in the header. It also makes the code a little more concise.

Reply:
For me, readability trumps compilation time. Why does it have to be a template? And what does it do?

Reply:
Currently we have `stringSetting()`, `sizetSetting()` and `boolSetting()`. Each for reading setting values of the respective type. My new setting falls into `stringSetting()` category but it has a limited set of choices so naturally I want to validate that. I think this is a common pattern so I thought I'd define `choiceSetting()` that uses `stringSetting()` underneath but also gets a list of valid values and validates the setting.

The natural extension is to define `enumSetting()` which also converts the string value to an enum for you. For that we need a template though.